New spack packages for the spack-repo

julianhoersch opened the pull request:
Added 8 new packages of which two are only first versions that do not work yet.

working:
- py-dials-data (for dials)
- py-numexpr 
- py-procrunner
- py-pyfai
- py-silx
- py-transformations

not working:
- py-dials (first version of dials - it  is not working yet)
- py-extra-foam (added dependencies - not working yet)

Updated spack.yaml
Updated readme

julianhoersch committed:
Added first versions of new packages to the repo

Added packages: py-dials (not working), py-dials-data (working), py-numexpr (working), py-procrunner (working), py-silx (working)

julianhoersch committed:
Added first versions of new packages to the repo

Added packages: py-dials (not working), py-dials-data (working), py-numexpr (working), py-procrunner (working), py-silx (working)

julianhoersch committed:
Added test dependencies

julianhoersch committed:
Added test dependencies

julianhoersch committed:
Added dependencies - not working yet

julianhoersch committed:
Added dependencies - not working yet

julianhoersch committed:
Added v0.20.0

julianhoersch committed:
Added v0.20.0

julianhoersch committed:
Update spack.yaml

julianhoersch committed:
Update spack.yaml

julianhoersch committed:
Update readme.md

Added new packages

julianhoersch committed:
Update readme.md

Added new packages

julianhoersch committed:
Update readme.md

Removed duplicate

julianhoersch committed:
Update readme.md

Removed duplicate

julianhoersch committed:
Update package.py

removed space

julianhoersch committed:
Update package.py

removed space

julianhoersch committed:
Update readme.md

Added transformations package and updated pyFAI dependencies

julianhoersch committed:
Update readme.md

Added transformations package and updated pyFAI dependencies

julianhoersch committed:
Added new packages

julianhoersch committed:
Added new packages

julianhoersch committed:
Added comment

julianhoersch committed:
Added comment

julianhoersch committed:
Removed GUI dependencies, added comments

GUI dependencies make packaging complicated

julianhoersch committed:
Removed GUI dependencies, added comments

GUI dependencies make packaging complicated

julianhoersch committed:
Added new package (dependency for py-pyFAI)

julianhoersch committed:
Added new package (dependency for py-pyFAI)

julianhoersch committed:
Merge branch 'master' of github.com:julianhoersch/spack-repo

julianhoersch committed:
Added Comment

julianhoersch committed:
Added Comment

julianhoersch committed:
Added comment

julianhoersch committed:
Added comment

julianhoersch committed:
Update package.py

julianhoersch committed:
Update package.py

julianhoersch committed:
Update package.py

julianhoersch committed:
Update package.py

julianhoersch committed:
Added comment

julianhoersch committed:
Added comment

julianhoersch committed:
Added comment

julianhoersch committed:
Added comment

julianhoersch committed:
Update package.py

julianhoersch committed:
Update package.py

julianhoersch committed:
Changed comment

julianhoersch committed:
Changed comment

julianhoersch committed:
Updated comments

julianhoersch committed:
Updated comments

julianhoersch committed:
Added maintainers

julianhoersch committed:
Added maintainers

julianhoersch committed:
Update readme.md

julianhoersch committed:
Update readme.md

RobertRosca commented on packages/py-dials-data/package.py:
```diff
@@ -0,0 +1,70 @@
+    depends_on('python@3.6:', type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
+    depends_on('py-importlib-resources@1.1.0:')
+    depends_on('py-pytest@6.1.2') # in install_requires
```
```suggestion
    depends_on('py-pytest@6.1.2', type=('test', 'build', 'run'))  # Required for tests, but also a package requirement
```

Doing this doesn't really have an effect (as if it's a build/run dependency it'll be installed before the tests already) but in this case it helps make it a bit more obvious that pytest is required for all three. Worth doing as that's a bit non-standard.

julianhoersch replied:
Yes, that should make it more obvious.

RobertRosca commented on packages/py-dials/package.py:
```diff
@@ -0,0 +1,126 @@
+    depends_on('py-six')
+    
+    # tests
+    depends_on('py-pytest-runner', type='test') # add type=test? only needed during development
```
Hmm, for standard Python packages it's not really worth using Spack for development, so feel free to ignore dev dependencies.

julianhoersch replied:
okay, as the dials package must be probably completely reworked to make it work I'll leave it like that for now.

RobertRosca commented on packages/py-extra-foam/package.py:
```diff
@@ -0,0 +1,86 @@
+    depends_on('py-msgpack-numpy@0.4.4:')
+    depends_on('py-pyzmq@17.1.2:')
+    depends_on('py-pyfai@0.17.0:')
+    depends_on('py-pyqt5@5.13.2') # needs higher compiler version as default on maxwell (gcc@4.8.5) -> gcc@:5.0.999
```
This is (in 'standard' deployments) fine, as I bump our compiler version up to `gcc@10.2.0`.

RobertRosca commented on packages/py-pyfai/package.py:
```diff
@@ -24,35 +24,49 @@
+    depends_on('py-cython')
     depends_on('py-wheel')
+    depends_on('py-transformations')
+    #depends_on('py-nbsphinx') commited to spack but not in spack repo yet
```
Spinx should (I assume) only be a requirement when building docs, so it's fine to leave it off.

RobertRosca reviewed (commented):
Sweet! Just went through the PR on GitHub and left a few comments to answer the questions you wrote, I'll go through this in more detail today!

RobertRosca committed:
Run workflow generation script

RobertRosca committed:
Fix pytest call

julianhoersch committed:
Merge branch 'master' of github.com:julianhoersch/spack-repo

julianhoersch committed:
Changed type of numpy dependency

Changed from 'build' to 'build' and 'test'

julianhoersch committed:
Fix syntax error

julianhoersch committed:
Added pytest

julianhoersch committed:
Corrected syntax error

julianhoersch committed:
Added pytest

julianhoersch committed:
Added py-fabio

julianhoersch committed:
Added pytest to py-numexpr

julianhoersch committed:
Added pytest to py-silx

julianhoersch committed:
Change python to python3

Try to fix errors during testing

julianhoersch committed:
Change python to python3 in py-extra-data

Trying to fix errors during CI testing

julianhoersch committed:
Changed test method to doctest

Using the test method defined in the transformations.py source code.

julianhoersch committed:
Changed test method to custom

Changed 'pytest' to the test method defined in the silx package 'silx.test.run_tests()'

julianhoersch committed:
Merge branch 'master' of github.com:julianhoersch/spack-repo

julianhoersch committed:
Undo python3 changes

julianhoersch committed:
added install_time_test_callbacks

julianhoersch committed:
Generated workflow for py-fabio

julianhoersch committed:
removed pytest

julianhoersch committed:
removed pytest

julianhoersch committed:
added spack load {package}

julianhoersch committed:
rerun workflow generation script

julianhoersch committed:
test, added spack load py-numpy

julianhoersch committed:
removed load {PACKAGE}

julianhoersch committed:
rerun workflow script

julianhoersch committed:
added spack load

julianhoersch committed:
Update SHA256 in extra-data from 1.2.0 to 1.3.0.

julianhoersch committed:
Update SHA256 in extra-data from 1.2.0 to 1.3.0

julianhoersch committed:
Update SHA256 in extra-data from 1.2.0 to 1.3.0

julianhoersch committed:
Delete test.py

pushed by accident

julianhoersch committed:
Merge branch 'panosc-eu:master' into master

julianhoersch committed:
add versions change dependecy types since test were failing otherwise

julianhoersch committed:
added versions and test dependencies

julianhoersch committed:
update version and pytest

julianhoersch committed:
removed boilerplate and comments

julianhoersch committed:
update crystel package

julianhoersch committed:
add new version, update test and clean package file

julianhoersch committed:
add versions, and remove unnecessary dependencies

julianhoersch committed:
added version, changed test directroy

julianhoersch committed:
added version, changed test directroy

julianhoersch committed:
removing boilerplate, change wdir of test

julianhoersch committed:
removed boilerplate, unnecessary imports, cleanup

julianhoersch committed:
add vagabond

julianhoersch committed:
add mkl support

julianhoersch committed:
correct syntax error

julianhoersch committed:
work in progress

julianhoersch committed:
changes in test method

julianhoersch committed:
updates

julianhoersch committed:
Merge remote-tracking branch 'upstream/master'

Conflicts:
	.docker/opt/spack/etc/spack/spack.yaml
	packages/vagabond/package.py

	test/

julianhoersch committed:
Merge branch 'master' of github.com:julianhoersch/spack-repo

RobertRosca committed:
Merge branch 'master' into master
